lib/ulid/rails.rb

```diff
@@ -13,6 +12,10 @@ module Rails
 
     class_methods do
       def ulid(column_name, primary_key: false, auto_generate: nil)
+        if "#{ActiveRecord::VERSION::MAJOR}.#{ActiveRecord::VERSION::MINOR}" == "4.2"
+          self.primary_key = column_name if primary_key
+        end
```

Comment:
I am not sure how often this `.ulid` method is called. Is it a problem to have the string concatenation (and allocation of `"4.2"` each time) during runtime?

Reply:
it is called everytime the model needs to be aware of its attribute type. mainly when setting and saving to db.